File: Programs/Part-A/Program-7/Java/app/src/main/java/com/demo/code/MainActivity.java
package com.demo.code;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.speech.tts.TextToSpeech;
import android.widget.Button;
import android.widget.EditText;

import org.w3c.dom.Text;

import java.util.Locale;

public class MainActivity extends AppCompatActivity {

    private EditText editTextTextPersonName;
    private Button btnConvertId;

    private TextToSpeech textToSpeech;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        findViews();
        setOnClickListeners();
        initilizeTextToSpeech();
    }

    private void initilizeTextToSpeech() {
        textToSpeech = new TextToSpeech(getApplicationContext(), i -> {
            if(i!=TextToSpeech.ERROR){
                // To Choose language of speech
                textToSpeech.setLanguage(Locale.UK);
            }
        });
    }

    private void findViews() {
        editTextTextPersonName = findViewById(R.id.editTextTextPersonName);
        btnConvertId = findViewById(R.id.btnConvertId);
    }

    private void setOnClickListeners() {
        btnConvertId.setOnClickListener(v -> {

            String inputEntered = editTextTextPersonName.getText().toString();

            textToSpeech.speak(inputEntered,TextToSpeech.QUEUE_FLUSH,null);
        });
    }

}

File: Programs/Part-A/Program-7/Kotlin/app/src/main/java/com/demo/code/MainActivity.kt
package com.demo.code

import android.os.Bundle
import android.speech.tts.TextToSpeech
import android.view.View
import android.widget.Button
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity
import com.demo.code.databinding.ActivityMainBinding
import java.util.*

class MainActivity : AppCompatActivity() {

    private var textToSpeech: TextToSpeech? = null
    private lateinit var binding: ActivityMainBinding


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        setOnClickListeners()
        initilizeTextToSpeech()
    }

    private fun initilizeTextToSpeech() {
        textToSpeech = TextToSpeech(applicationContext) { i: Int ->
            if (i != TextToSpeech.ERROR) {
                // To Choose language of speech
                textToSpeech?.language = Locale.UK
            }
        }
    }

    private fun setOnClickListeners() {
        binding.apply {
            btnConvertId.setOnClickListener {
                val inputEntered = editTextTextPersonName.text.toString()
                textToSpeech?.speak(inputEntered, TextToSpeech.QUEUE_FLUSH, null)
            }
        }
    }
}